### PR TITLE
Updated to use SSLHost

### PR DESCRIPTION
**Pull request by PaulLodge**

### CODE REVIEW

File: core/src/main/groovy/noe/server/Tomcat.groovy [hunk 1]
```diff
@@ -669,14 +669,18 @@ class Tomcat extends ServerAbstract implements WorkerServer {
     def nl = platform.nl
     def dummyComment = '<!-- Define an AJP 1.3 Connector on port 8009 -->'
     def enableOpenSsl = '<Connector port="' + this.mainHttpsPort.toString() + '" SSLEnabled="true"' + nl +
-        'maxThreads="200" scheme="https" secure="true"' + nl +
-        'SSLCertificateFile="' + this.sslCertificate + '"' + nl +
-        'SSLCertificateKeyFile="' + this.sslKey + '"' + nl +
-        'SSLPassword="' + this.sslKeystorePassword + '"' + nl
+            'maxThreads="200" scheme="https" secure="true"' + nl
+    'maxThreads="200" scheme="https" secure="true"' + nl +
```
Comment (sgala) [inline core/src/main/groovy/noe/server/Tomcat.groovy:671]: Both lines are duplicated except for whitespace

**Review by sgala — APPROVED** [review]
I just noticed the duplicate line I commented on

File: core/src/main/groovy/noe/server/Tomcat.groovy (hunk 1 above)
Comment (PaulLodge) [reply core/src/main/groovy/noe/server/Tomcat.groovy:671]: thanks I hadn't noticed it

File: core/src/main/groovy/noe/server/Tomcat.groovy [hunk 2]
```diff
@@ -641,47 +641,47 @@ class Tomcat extends ServerAbstract implements WorkerServer {
...
   void enableSslOpenSsl() {
     def nl = platform.nl
     def dummyComment = '<!-- Define an AJP 1.3 Connector on port 8009 -->'
     def enableOpenSsl = '<Connector port="' + this.mainHttpsPort.toString() + '" SSLEnabled="true"' + nl +
-        'maxThreads="200" scheme="https" secure="true"' + nl +
-        'SSLCertificateFile="' + this.sslCertificate + '"' + nl +
-        'SSLCertificateKeyFile="' + this.sslKey + '"' + nl +
-        'SSLPassword="' + this.sslKeystorePassword + '"' + nl
+            'maxThreads="200" scheme="https" secure="true"' + nl
+            'sslImplementationName="org.apache.tomcat.util.net.openssl.OpenSSLImplementation"' + nl
```
Comment (AdamKrajcik) [inline core/src/main/groovy/noe/server/Tomcat.groovy:671]: There seems to be some  + signs missing
```suggestion
            'maxThreads="200" scheme="https" secure="true"' + nl +
            'sslImplementationName="org.apache.tomcat.util.net.openssl.OpenSSLImplementation"' + nl +
```

File: core/src/main/groovy/noe/server/Tomcat.groovy [hunk 3]
```diff
@@ -641,47 +641,47 @@ class Tomcat extends ServerAbstract implements WorkerServer {
     this.updateConfReplaceRegExp('server.xml', '<Connector port="8009" protocol="AJP/1.3"', '<Connector port="' + Integer.valueOf(8009 + offset) + '" protocol="AJP/1.3"', true, true)
   }
 
-  @Deprecated
```
Comment (AdamKrajcik) [inline core/src/main/groovy/noe/server/Tomcat.groovy:644]: Not sure if we want to undeprecate this method, as it's using some updateConfReplaceRegExp() and a dummy comment to update the configuration.

**Review by AdamKrajcik — COMMENTED** [review]
Marking as comment

File: core/src/main/groovy/noe/server/Tomcat.groovy [hunk 4]
```diff
@@ -641,47 +641,47 @@ class Tomcat extends ServerAbstract implements WorkerServer {
...
-              '          SSLEnabled="true" maxThreads="150" scheme="https" secure="true"' + nl +
-              '          clientAuth="false" sslEnabledProtocols="TLSv1.1+TLSv1.2"' + nl +
-              '          keystoreFile="' + this.keystorePath + '"' + nl +
-              '          keystorePass="' + this.sslKeystorePassword + '"' + nl +
-              '          keystoreType="' + this.keystoreType + '"' + nl +
-              '          ciphers="' + DefaultProperties.FIPS_140_2_CIPHERS+ '" />'
-      }
+    def ciphers = ''
+    if(platform.isFips()) {
+      ciphers = 'sslEnabledProtocols="TLSv1.1+TLSv1.2" ciphers="' + DefaultProperties.FIPS_140_2_CIPHERS+ '"'
```
[the remark this answers is not in the document]
Comment (PaulLodge) [reply core/src/main/groovy/noe/server/Tomcat.groovy:652]: for now, I would keep things as they are

**Comment by AdamKrajcik** [comment]
I have a bit of sad news. This configuration won't work. See https://tomcat.apache.org/tomcat-10.1-doc/config/http.html#SSL_Support_-_SSLHostConfig

**Comment by PaulLodge** [comment]
which config there are 2, a little information maybe?

**Comment by AdamKrajcik** [comment]
Not up to date and small issues.

**Pull request closed without merging**